sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -120,76 +125,80 @@ def __init__(
         self._training_args = training_args if training_args is not None else {}
         self.compute_metrics = compute_metrics
         self._compute_metrics = compute_metrics
-        self._compute_metrics = compute_metrics
         self.deterministic = deterministic
         self.callbacks = callbacks
         self._callbacks = callbacks
 
     def _fit(self, y, X, fh):
-        # Load model and extract config
-        config = AutoConfig.from_pretrained(self.model_path)
-
-        # Update config with user provided config
-        _config = config.to_dict()
-        _config.update(self._config)
-        _config["num_dynamic_real_features"] = X.shape[-1] if X is not None else 0
-        _config["num_static_real_features"] = 0
-        _config["num_dynamic_real_features"] = 0
-        _config["num_static_categorical_features"] = 0
-        _config["num_time_features"] = 0 if X is None else X.shape[-1]
-
-        if hasattr(config, "feature_size"):
-            del _config["feature_size"]
-
-        if fh is not None:
-            _config["prediction_length"] = max(
-                *(fh.to_relative(self._cutoff)._values + 1),
-                _config["prediction_length"],
-            )
+        if self.model is None and self.model_path is not None:
+            # Load model and extract config
+            config = AutoConfig.from_pretrained(self.model_path)
```

Comment:
I would propose to move the model loading functionality into a private method to keep the fit method simpler. E.g. into a method `_load_model`